models.ts

```diff
@@ -9,10 +9,45 @@ class Peli {
 
 class PelisCollection {
   getAll(): Promise<Peli[]> {
-    return jsonfile("...laRutaDelArchivo").then(() => {
-      // la respuesta de la promesa
-      return [];
+    let all = jsonfile.readFile("./pelis.json").then((x) => x);
+
+    return all;
+  }
+
+  getById(id: number) {
+    let encontrarId = this.getAll().then((x) => x.find((i) => i.id == id));
+    return encontrarId;
+  }
+
+  search(options: any) {
+    let buscar = this.getAll();
+
+    if (options.title) {
+      let encontrarTitle = buscar.then((x) =>
+        x.filter((x) => x.title.includes(options.title))
+      );
+      buscar = encontrarTitle;
+    }
+
+    if (options.tag) {
+      let encontrarTag = buscar.then((x) =>
+        x.filter((x) => x.tags.includes(options.tag))
+      );
+      buscar = encontrarTag;
+    }
+
+    return buscar;
+  }
+
+  add(peli: Peli): Promise<Boolean> {
+    return this.getAll().then((x) => {
+      const comprobar = x.find((i: Peli) => i.id === peli.id);
```

Comment:
este problema ya lo resolviste en el método getById, lo podés reutilizar haciendo this.getById()